plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -462,24 +462,39 @@ private BrokerResponseNative submitBrokerQueryJson(ConnectorSession session, Pin
     public Iterator<BrokerResultRow> createResultIterator(ConnectorSession session, PinotQuery query, List<PinotColumnHandle> columnHandles)
     {
         BrokerResponseNative response = submitBrokerQueryJson(session, query);
-        return fromResultTable(response.getResultTable(), columnHandles);
+        return fromResultTable(response, columnHandles, query.getGroupByClauses());
     }
 
     @VisibleForTesting
-    public static ResultsIterator fromResultTable(ResultTable resultTable, List<PinotColumnHandle> columnHandles)
+    public static ResultsIterator fromResultTable(BrokerResponseNative brokerResponse, List<PinotColumnHandle> columnHandles, int groupByClauses)
     {
-        requireNonNull(resultTable, "resultTable is null");
+        requireNonNull(brokerResponse, "brokerResponse is null");
         requireNonNull(columnHandles, "columnHandles is null");
+        ResultTable resultTable = brokerResponse.getResultTable();
         String[] columnNames = resultTable.getDataSchema().getColumnNames();
         Map<String, Integer> columnIndices = IntStream.range(0, columnNames.length)
                 .boxed()
                 // Pinot lower cases column names which use aggregate functions, ex. min(my_Col) becomes min(my_col)
                 .collect(toImmutableMap(i -> columnNames[i].toLowerCase(ENGLISH), identity()));
         int[] indices = new int[columnNames.length];
+        int[] inverseIndices = new int[columnNames.length];
         for (int i = 0; i < columnHandles.size(); i++) {
             indices[i] = columnIndices.get(columnHandles.get(i).getColumnName().toLowerCase(ENGLISH));
+            inverseIndices[indices[i]] = i;
         }
-        return new ResultsIterator(resultTable, indices);
+        List<Object[]> rows = resultTable.getRows();
+        // If returning from a global aggregation (no grouping columns) over an empty table, NULL-out all aggregation function results except for `count()`
```

Comment:
cc: @martint This looks correct to me (I looked at most aggregation functions) but I'd appreciate your view if there are any other functions that behave like `count`?

Reply:
No, count is the only function defined in the SQL spec that has that behavior.